ui/src/app/job-list/table/table.component.ts

```diff
@@ -28,37 +33,40 @@ import {environment} from '../../../environments/environment';
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.css'],
 })
+@Injectable()
 export class JobsTableComponent implements OnInit {
   @Input() dataSource: DataSource<QueryJobsResult>;
   @Output() onJobsChanged: EventEmitter<QueryJobsResult[]> = new EventEmitter();
 
   private mouseoverJob: QueryJobsResult;
 
-  public additionalColumns: string[] = [];
+  public displayFields: DisplayField[];
   public selection = new SelectionModel<QueryJobsResult>(/* allowMultiSelect */ true, []);
   public jobs: QueryJobsResult[] = [];
 
   // TODO(alanhwang): Allow these columns to be configured by the user
-  displayedColumns = primaryColumns.slice();
+  displayedColumns: string[] = ["Job", "Details"];
```

Comment:
Is this array still needed? Could displayFields subsume this?

Reply:
This determines all the columns displayed on screen, including the default "Job" and details drop down columns. One issue is that displayFields is not always immediately available, so these are the default columns shown until the capabilities endpoint returns. We could probably  have a function which does this but it seems simpler to just keep track of it in a variable.